src/openrct2/world/park.c

```diff
@@ -477,6 +477,22 @@ static sint32 park_calculate_guest_generation_probability()
 	return probability;
 }
 
+void get_nearest_peep_spawn(rct2_peep_spawn *spawn, sint32 x, sint32 y)
+{
+	sint32 nearestDist = INT32_MAX;
+	sint32 nearestIndex = MAX_PEEP_SPAWNS;
+	for (sint32 i = 0; i < MAX_PEEP_SPAWNS; i++) {
+		sint32 dx = gPeepSpawns[i].x - x;
+		sint32 dy = gPeepSpawns[i].y - y;
+		sint32 dist = dx * dx + dy * dy;
```

Comment:
aaaand i just now realize that peep_path_find_entering_park has a usable distance algorithm code i can use for peep_path_find_leaving_park.... should i still store it in a get_nearest_peep_spawn, or not worry about having it in a seperate function?